tempodb/encoding/vparquet/block_search_test.go

```diff
@@ -218,8 +219,42 @@ func TestBackendBlockSearch(t *testing.T) {
 	}
 }
 
-func makeBackendBlockWithTraces(t *testing.T, trs []*Trace) *backendBlock {
+func TestBackendBlockSearchTags(t *testing.T) {
+	traces, attrs := makeTraces()
+	block := makeBackendBlockWithTraces(t, traces)
+
+	cb := func(s string) {
+		_, exists := attrs[s]
+		require.True(t, exists, s)
+		delete(attrs, s)
+	}
+
+	ctx := context.Background()
+	err := block.SearchTags(ctx, cb, defaultSearchOptions())
+	require.NoError(t, err)
+
+	require.Empty(t, attrs)
```

Comment:
I'm finding the inverted nature of this test affects readability.  It seems like we want to assert that every tag is found, and only found once.   Could we collect the unique tags into another map and `require.Equal(t, attrs, foundAttrs)` ?

Reply:
reorganized the test